### ISSUE TITLE
Make Mastodon More Semantic Web

### ISSUE THREAD

**Issue by marrus-sh** [glitch-soc/mastodon#43]
This is better for accessibility and better for life.  
Right now Mastodon uses `<div>` for everything. Gross.  
It *should* use `<header>`, `<nav>`, `<footer>`, `<aside>`, `<section>`, `<main>` where appropriate.  
Statuses *should* be wrapped in `<article>` as they are standalone potentially-rebloggable content.

Also Mastodon should really use `<b>` instead of `<strong>` unless it is intending to be shouting.

**Comment by yiskah** [comment]
Err so <div> is used in a lot of the various plugins and the like. Particularly when it comes to JSX which looks like HTML but is not HTML and does not work the same. Changing things to these other tags would likely break a lot of stuff

Also using <b> instead of <strong> is a violation of WCAG. afaik there's no reason to use <b> when you could use <strong>

**Comment by marrus-sh** [comment]
`<strong>` should only be used for content which has strong importance. Using `<strong>` for text which does not have strong importance is against spec and `<b>` should be used instead. (Display names are currently rendered using `<strong>`, for example, which is an improper use of the element.)

**Comment by Chronister** [comment]
my understanding of best practices here would be to introduce a span -- a
general purpose inline element -- with a class that has font-weight
applied, rather than use the deprecated <b> tag or a somewhat tangentially
related <strong>, since as Gô says, the username, while important, is not
exactly "strongly emphasized" text in that sense.

**Comment by MightyPork** [comment]
from what I read, b and i are not deprecated in HTML5 anymore, instead b should be used for something that's meant to be easy to find in the text, and i for italics, as in for latin names etc.

source: https://html.spec.whatwg.org/multipage/text-level-semantics.html#the-i-element

span with no semantic meaning does not seem appropriate to me

**Comment by yiskah** [comment]
Oh I didn't realize it was being used for usernames. Yes in that case <strong> isn't quite right.

**Comment by marrus-sh** [comment]
the big problem with it is that it suggests that people's display names (rendered with `<strong>`) is more important than their username (rendered with `<span>` iirc) which is *backwards* imo.

but yeah that's what I was talking about. not eliminating `<strong>` everywhere, but it's a semantic element and shouldn't be used for styling purposes.

**Comment by yiskah** [comment]
Right right. It should be used for "Mastodon is **decentralized**" but not "**Go** posted"

Also I disagree I think a display name is more important than a username. "@icosahedron.website" is less important information than like, my name. I know ppl make jokes 'n stuff but the username is someone's like... address? Whereas display name is what they're going by. Akin to email, it's more important that Sheryl Smith has emailed me than that csmith@company.com emailed me

**Comment by MightyPork** [comment]
I have to disagree that display name is more important than username (yeah, that exact argument, ok ok)

Thing is, some people use unicode soup (looks at kibi) and similar funsies there and then you can't tell who it is when they eg. boost or fave your toot. Personally I'd change StatusPrepend to show the username there, for example.

**Comment by MightyPork** [comment]
looks like sorin started working in this direction: https://github.com/tootsuite/mastodon/pull/4369

**Comment by marrus-sh** [comment]
yeah i noticed upstream is doing this now~ we can probably leave it to them, then
